### PR TITLE
✨(frontend) add document search

### PR DESCRIPTION
**Pull request by PanchoutNathan**
## Purpose

Added document search functionality

Capture d’écran 2024-12-09 à 18 02 40

### CODE REVIEW

File: src/frontend/apps/impress/src/features/docs/doc-search/components/DocSearchModal.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import { InView } from 'react-intersection-observer';
+import { css } from 'styled-components';
+import { useDebouncedCallback } from 'use-debounce';
+
+import { Box } from '@/components';
+import {
+  QuickSearch,
+  QuickSearchData,
+} from '@/components/quick-search/QuickSearch';
+import { QuickSearchGroup } from '@/components/quick-search/QuickSearchGroup';
```
Comment (AntoLC) [inline src/frontend/apps/impress/src/features/docs/doc-search/components/DocSearchModal.tsx:15]:
```suggestion
import {
  QuickSearch,
  QuickSearchData,
  QuickSearchGroup
} from '@/components/quick-search/';
```

File: src/frontend/apps/impress/src/components/quick-search/QuickSearchItemContent.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+import { Box } from '../Box';
+
+export type QuickSearchItemContentProps = {
+  alwaysShowRight?: boolean;
+  left: ReactNode;
+  right?: ReactNode;
+};
+
+export const QuickSearchItemContent = ({
```
Comment (AntoLC) [inline src/frontend/apps/impress/src/components/quick-search/QuickSearchItemContent.tsx:13]: Will you need it ?

File: src/frontend/apps/impress/src/features/docs/doc-search/components/DocSearchModal.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                  alt={t('No active search')}
+                />
+              </Box>
+            )}
+            {search && docsData.elements.length > 0 && (
+              <QuickSearchGroup
+                onSelect={handleSelect}
+                group={docsData}
+                renderElement={(doc) => <DocSearchItem doc={doc} />}
+              />
```
Comment (AntoLC) [inline src/frontend/apps/impress/src/features/docs/doc-search/components/DocSearchModal.tsx:102]: Could be improved, DocSearchItem rerenders even when docsData is the same. For this usecase it is ok I think.

File: src/frontend/apps/impress/src/components/quick-search/QuickSearchGroup.tsx
Comment (AntoLC) [inline src/frontend/apps/impress/src/components/quick-search/QuickSearchGroup.tsx:1]: I don't understand to much the hovering.

[scrnli_2WLwbyOWrlPuT5.webm](https://github.com/user-attachments/assets/0215939d-9d85-45ec-b331-21c264319b76)

**Review by AntoLC — COMMENTED** [review]
It's clean, works well.

I find QuickSearch a bit complicated after for what you really need, an input.
I think I would have encapsulate the DataGrid in a modal, and just have a modal DataGridSearchModal, instead recreating part of the logic. By encapsulate the datagrid, you will let easily the door open to add actions, maybe you want to search some docs to delete them in bulk or stuff like that.

File: src/frontend/apps/impress/src/hook/useCmdK.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
```
[the remark this answers is not in the document]
Comment (PanchoutNathan) [reply src/frontend/apps/impress/src/hook/useCmdK.tsx:1]: Auto import ... my bad

File: src/frontend/apps/impress/src/features/docs/doc-search/assets/empty-search.svg
[the remark this answers is not in the document]
Comment (PanchoutNathan) [reply src/frontend/apps/impress/src/features/docs/doc-search/assets/empty-search.svg:1]: to delete obviously

File: src/frontend/apps/impress/src/components/quick-search/QuickSearchItemContent.tsx (hunk 2 above)
Comment (PanchoutNathan) [reply src/frontend/apps/impress/src/components/quick-search/QuickSearchItemContent.tsx:13]: Yes i forgot to use it, i duplicate the right content logic inside DocSearchItem

File: src/frontend/apps/impress/src/components/quick-search/QuickSearchGroup.tsx
Comment (PanchoutNathan) [reply src/frontend/apps/impress/src/components/quick-search/QuickSearchGroup.tsx:1]: fixed

File: src/frontend/apps/impress/src/features/docs/doc-search/components/DocSearchModal.tsx (hunk 3 above)
Comment (PanchoutNathan) [reply src/frontend/apps/impress/src/features/docs/doc-search/components/DocSearchModal.tsx:98]: yeah i think it's ok for this use case too

**Review by AntoLC — APPROVED** [review]

**Pull request merged**